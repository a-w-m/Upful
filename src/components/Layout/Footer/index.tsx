import React from "react"
import {
  FooterContainer,
  FooterNav,
  CopyrightWrapper,
  Address,
  FooterLink,
} from "./styled"
import { useStaticQuery, graphql } from "gatsby"
import { C } from "../../../interfaces"

const Footer: React.FC<{}> = () => {
  
  interface Data {
    allMarkdownRemark: {
      edges: {
        node: {
          frontmatter: {
            title: string
          }
          fields: {
            slug: string
          }
        }
      }[]
    }
    site:C.FooterLinks["site"]
  }

  const footerData: Data = useStaticQuery(footerQuery)

  return (
    <FooterContainer>
      <FooterNav aria-label = "Footer">
        {footerData.site.siteMetadata.menuLinks.categories.map(category => {
          return (
            <FooterLink to={category.slug} key={category.name}>
              {category.name}
            </FooterLink>
          )
        })}
        {footerData.allMarkdownRemark.edges.map(page => {
          return (
            <FooterLink
              to={page.node.fields.slug}
              key={page.node.frontmatter.title}
            >
              {page.node.frontmatter.title.toLowerCase()}
            </FooterLink>
          )
        })}
      </FooterNav>
      <Address>
        <a href={footerData.site.siteMetadata.email}>{footerData.site.siteMetadata.email}</a>
      </Address>
      <CopyrightWrapper>
        Â© Upful {new Date(Date.now()).getFullYear()} Made with full Irations
      </CopyrightWrapper>
    </FooterContainer>
  )
}

export const footerQuery = graphql`
  {
    allMarkdownRemark(filter: { fileAbsolutePath: { regex: "/pages/" } }) {
      edges {
        node {
          fields {
            slug
          }
          frontmatter {
            title
          }
        }
      }
    }
    site {
      siteMetadata {
        email
        menuLinks {
          categories {
            name
            slug
          }
        }
      }
    }
  }
`

export default Footer
