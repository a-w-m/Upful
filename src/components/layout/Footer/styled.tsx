import styled from "styled-components"
import device from "../mediaQuery"

export const FooterContainer = styled.footer`
  grid-area: footer;
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  background-color: #153325;
  padding: 0.5rem;

  & > nav {
    margin: 0.5rem;
  }

  & > div {
    margin: 0.5rem;
  }

  @media ${device.laptop}{
    background: var(--main-brown);
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
`

export const FooterNav = styled.nav`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;


  a {
    font-size: 1.2rem;
    text-decoration: none;
    line-height: 2.1rem;
    color: var(--link-text-color);


  }



  @media ${device.laptop}{
    flex-direction: row;
    flex-basis: 100%; 
    flex-shrink: 0;
    gap: 2rem;
  }
`

export const ContactWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

export const TermsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  a {
    font-size: 1.2rem;
    color: var(--link-text-color);
    text-decoration: none;
    line-height: 1rem;
  }
`

export const CopyrightWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  color: var(--link-text-color);
`

export const Heading = styled.h5`
  color: #fff;
  font-size: 1.5rem;
`
export const Address = styled.address`
  text-align: center;
  color: var(--link-text-color);
  font-style: normal;
  line-height: 2.1rem;

  a {
    text-decoration: underline;
      color: var(--link-text-color);
  }

  @media ${device.laptop}{
    flex-basis: 100%; 
  }
`
