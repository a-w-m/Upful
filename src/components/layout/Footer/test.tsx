import React from 'react'
import {render, screen} from '@testing-library/react'
import Footer from "."


describe("Footer Component", ()=>{

    test("matches component", ()=>{
        const {container} = render(<Footer />)
        expect(container.firstElementChild).toMatchSnapshot()

    })

})