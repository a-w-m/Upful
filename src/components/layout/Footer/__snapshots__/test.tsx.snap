// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches component 1`] = `
<footer
  class="sc-bdfBwQ eUkNkO"
>
  <nav
    class="sc-gsTCUz bClOqD"
  >
    <h5
      class="sc-jSgupP hSPQha"
    >
      Categories
    </h5>
    <a
      href="/about/"
    >
      About Us
    </a>
    <a
      href="/clothing/"
    >
      Clothing
    </a>
    <a
      href="/home/"
    >
      Home
    </a>
    <a
      href="/accessories/"
    >
      Accessories
    </a>
  </nav>
  <div
    class="sc-dlfnbm crRguJ"
  >
    <h5
      class="sc-jSgupP hSPQha"
    >
      Contact
    </h5>
    <address
      class="sc-gKsewC eDEVly"
    >
      StoreFront
      <br />
      555 Santa Monica Blvd.
      <br />
      Los Angeles, CA 90210
    </address>
    <address
      class="sc-gKsewC eDEVly"
    >
      <a
        href="mailto:howdy@storefront.com"
      >
        howdy@storefront.com
      </a>
    </address>
  </div>
  <div
    class="sc-hKgILt gUssEG"
  >
    <h5
      class="sc-jSgupP hSPQha"
    >
      Terms & Conditions
    </h5>
    <a
      href="/returns"
    >
      Returns
    </a>
  </div>
  <div
    class="sc-eCssSg fHHycj"
  >
    Â© StoreFront 
    2019
  </div>
</footer>
`;
